dotnet/samples/KernelSyntaxExamples/Skills/SegmentSkill.cs

```diff
@@ -0,0 +1,121 @@
+    /// {{file.readAsync $path }} => "hello world"
+    /// </example>
+    /// <param name="path"> Source file </param>
+    /// <returns> File content </returns>
+    [SKFunction, Description("Create a segment using prompt and parsing prompt")]
+    public async Task<string> CreateSegment([Description("Name of the segment.")] string segmentname,
+        [Description("Name of the segment definition. Some of the examples are FirstLoginDateFilter, LastLoginDateFilter, LocationFilter.")] string segmentdefinition,
+        [Description("Name of the segment comparison. Some of the examples are GreaterThan, LessThan, Equals.")] string segmentcomparison,
+        [Description("Value of the segment comparison. Some of the examples are 2023-08-01, India, Australia, Kenya.")] string segmentcomparisonvalue
+        )
+    {
+        //ToDo: Create payload json using Playfab dlls/sdk
+        // Set properties to create a Segment using swagger.json
+        ContextVariables contextVariables = new ContextVariables();
+        contextVariables.Set("content_type", "application/json");
+        contextVariables.Set("server_url", TestConfiguration.PlayFab.Endpoint);
+        string segmentPayload = GetSegmentPayload(segmentname, segmentdefinition, segmentcomparison, ref segmentcomparisonvalue);
+
+        contextVariables.Set("content_type", "application/json");
+        contextVariables.Set("payload", segmentPayload);
+        var kernel = new KernelBuilder().WithLogger(ConsoleLogger.Logger).Build();
+        using HttpClient httpClient = new();
+        var playfabApiSkills = await GetPlayFabSkill(kernel, httpClient);
+
+        // Run operation via the semantic kernel
+        var result2 = await kernel.RunAsync(contextVariables, playfabApiSkills["CreateSegment"]);
+
+        Console.WriteLine("\n\n\n");
+        var formattedContent = JsonConvert.SerializeObject(JsonConvert.DeserializeObject(result2.Result), Formatting.Indented);
+        Console.WriteLine("CreateSegment playfabApiSkills response: \n{0}", formattedContent);
+
+        return $"Segment {segmentname} created with segment definition {segmentdefinition}";
+    }
+
+    private static string GetSegmentPayload(string segmentname, string segmentdefinition, string segmentcomparison, ref string segmentcomparisonvalue)
+    {
+        string segmentPayload = "{\n  \"SegmentModel\": {\n \"Name\": \"<SegmentName>\",\n \"SegmentOrDefinitions\": [\n {\n \"SegmentAndDefinitions\": [\n {\n \"<SegmentDefinition>\": {\n \"LogInDate\": \"<SegmentComparisonValue>T00:00:00Z\",\n \"Comparison\": \"<SegmentComparison>\"\n }\n }\n ]\n }\n ]\n }\n  }";
+        string locationPayload = "{\n  \"SegmentModel\": {\n \"Name\": \"<SegmentName>\",\n \"SegmentOrDefinitions\": [\n {\n \"SegmentAndDefinitions\": [\n {\n \"<SegmentDefinition>\": {\n \"CountryCode\": \"<SegmentComparisonValue>\",\n \"Comparison\": \"<SegmentComparison>\"\n }\n }\n ]\n }\n ]\n }\n  }";
+
+        if (segmentdefinition == "LocationFilter")
```

Comment:
>LocationFilter [](http://example.com/codeflow?start=34&length=14)

With proper documentation this mapping steps won't be needed, and the model should know to return country codes (instead of country names) for that segment definition

Reply:
Yes. I think this comment is the same as before.